testsuite/features/support/custom_formatter.rb

```diff
@@ -4,20 +4,42 @@
 module CustomFormatter
   # Extend the Cucumber Pretty Formatter and prepends the feature name on each step
   module PrependsFeatureName
+    # A class method that takes a formatter class as an argument.
     def self.formatter(formatter_class)
       Class.new(formatter_class) { include PrependsFeatureName }
     end
 
+    ##
+    # It takes the feature file name and stores it in a variable called @current_feature_name
+    #
+    # Args:
+    #   feature: The feature object
     def before_feature(feature)
       @current_feature_name = feature.location.file
       super(feature)
     end
 
+    ##
+    # It takes the keyword, prepends the feature name to it, and then calls the super function
+    #
+    # Args:
+    #   keyword: The keyword of the step (Given, When, Then, etc)
+    #   step_match: The step match object
+    #   status: :passed, :failed, :undefined, :skipped, :pending, :exception
+    #   source_indent: The indentation of the step definition
+    #   background: true if the step is in a background
+    #   file_colon_line: "features/step_definitions/my_steps.rb:7"
```

Comment:
@srbarrios do you know what this is supposed to be? it's not clear to me :sweat_smile:

Reply:
This method prepend the feature name into the step name, we do that way to then when we look at the console output logs, we can find easily to what feature this step is related. (This was made at the moment we enabled two years ago the parallel tests and it was hard to follow the mixed console logs)

Reply:
I clarified those steps.